### PR TITLE
Introduce test for basic type string

### PR DESCRIPTION
**Pull request by yjaeseok**
I would like to add test for string type,
This is initial commit. After that I hope to add more test case.

### CODE REVIEW

**Comment by NickNaso** [comment]
Hi @yjaeseok your start is good. Improve the work and when you want a feedback just ping here.

File: test/basic_types/string.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+const buildType = process.config.target_defaults.default_configuration;
+const assert = require('assert');
+
+test(require(`../build/${buildType}/binding.node`));
+test(require(`../build/${buildType}/binding_noexcept.node`));
+
+function test(binding) {
+	const testString = binding.basic_types_string.createUtf8String("κόσμε");
+	assert.strictEqual(testString, "κόσμε");
+}
```
Comment (mhdawson) [inline test/basic_types/string.js]: Assuming this is still a work in progress as its not calling createUtf16String yet, but good start.   Let us know when you think its complete and ready to review.

Comment (yjaeseok) [reply test/basic_types/string.js]: Thank you for review ~
Yes, I am still a work in progress +_+//

I am considering how to make test for String type.
I need to understand String for ECMA script more.

**Comment by yjaeseok** [comment]
This issue has been pending too long, So I would like to update it later, once I add the basic test first.
@NickNaso @mhdawson

File: test/basic_types/string.js (hunk 1 above)
Comment (gabrielschulhof) [reply test/basic_types/string.js]: I think we can resolve this now, right? There is a call to `createUtf16String` now. *That* is what you were referring to, right, @mhdawson?

**Comment by mhdawson** [comment]
@yjaeseok sorry for the very late reply.  I think you were saying you think we should land what you have now and then you will build on it.  Is that right?

**Comment by yjaeseok** [comment]
@mhdawson I am sorry too, for the very late reply. These days I cannot anything for node-addon-api, because taking care of my sons. T.T..

Anyway I would like to merge this PR, and then keep contribution for node-addon-api asap.
Thanks !

@gabrielschulhof Thanks.

**Comment by gabrielschulhof** [comment]
CI:

| version | job |
|--- |--- |
| v6.x | https://ci.nodejs.org/job/node-test-node-addon-api/1597/ |
| v7.x |  |
| v8.x |  |
| v9.x |  |
| v10.x |  |
| v11.x |  |
| v12.x-nightly |  |

**Comment by mhdawson** [comment]
@yjaeseok are you going to be able to get back to this in the next few months?  If not I'm thinking we should close until you have more time.

**Comment by yjaeseok** [comment]
@mhdawson Could I close this issue until I have time ?

**Comment by mhdawson** [comment]
@joyeecheung yes that would be good.

**Pull request closed without merging**